tests/serialized_test_data_generation/Makefile

```diff
@@ -1,69 +1,138 @@
-
-pack_test_data:
-	if [ ! -d $(TEST_DATA_HOST)/$(TEST_DATA_TARFILE) ]; then \
-	cd $(TEST_DATA_HOST) && tar  --exclude=input.nml --exclude=*.json  --exclude=$(TEST_DATA_TARFILE) \
-	-czf $(TEST_DATA_TARFILE) . ;fi 
-
-	for number in 0 1 2 3 4 5 6 7 8 9 ; do \
-		cd $(TEST_DATA_HOST) && rm -f Gen*rank$$number*.dat ; \
-	done
-unpack_test_data:
-	if [ -f $(TEST_DATA_TARPATH) ]; then \
-	cd $(TEST_DATA_HOST) && tar -xf $(TEST_DATA_TARFILE) && \
-	rm $(TEST_DATA_TARFILE); fi
-
-push_test_data:
-	echo $(LOOKUP_BUCKET)
-	gsutil -m rsync $(TEST_DATA_HOST) $(TEST_DATA_BUCKET)	
-
-generate_and_push_data: setup_rundir run_model_make_data pack_test_data push_test_data
-
-generate_coverage: install_deps
-	rm -rf coverage
-	EXPERIMENT=fv3config_coverage $(MAKE) setup_rundir 	
-	cd $(FORTRAN_DIR) && DOCKER_BUILDKIT=1 $(MAKE) build_coverage
-	mkdir coverage
-	docker run -it --rm --network host -v $(PWD)/coverage:/coverage -v $(CWD)/rundir/fv3config_coverage:$(RUNDIR_CONTAINER) $(GCR_URL)/fv3gfs-compiled:gcov bash -c "pip install gcovr; cd /coverage; mkdir physics; cd physics; gcovr -d -r /FV3/gfsphysics --html --html-details -o index.html; cd ../; mkdir dycore; cd dycore; gcovr -d -r /FV3/atmos_cubed_sphere --html --html-details -o index.html"
-	@echo "==== Coverage ananlysis done. Now open coverage/dycore/index.html coverage/physics/index.html in your browser ===="
+# version number or tag (determines image tag and destination directory in cloud storage)
+# -convention: <tag>-<some large conceptual version change>.<serialization statement change>.<hotfix>
+# -version number should be increased when serialization data is expected to change
+# -omit tag (e.g. 7.1.1) for "operational" serialization data, use tag for experimental serialization data
+FORTRAN_VERSION ?= 7.1.1
 
+# docker container image setup (used for running model, see ../../README.md)
+GCR_URL = us.gcr.io/vcm-ml
+COMPILED_IMAGE = $(GCR_URL)/fv3gfs-compiled:$(FORTRAN_VERSION)-serialize-gt4pydev
+RUN_DIR_CONTAINER = /rundir
+DATA_DIR_CONTAINER = $(RUN_DIR_CONTAINER)/test_data
 
+# local host setup
+CWD=$(shell pwd)
+FORTRAN_DIR = $(CWD)/../../
+RUN_DIR_HOST = $(CWD)/rundir/$(EXPERIMENT)
+DATA_DIR_HOST = $(CWD)/data/$(EXPERIMENT)
+TARFILE_HOST = $(DATA_DIR_HOST)/dat_files.tar.gz
+TARFILE_INDEX_HOST = $(DATA_DIR_HOST)/tarfile_index.txt
+USER_ID_HOST = $(shell id -u)
+GROUP_ID_HOST = $(shell id -g)
+
+# cloud storage location for pushing generated data
+STORAGE_BUCKET = gs://vcm-fv3gfs-serialized-regression-data
+DATA_DIR_BUCKET = $(STORAGE_BUCKET)/$(FORTRAN_VERSION)/$(EXPERIMENT)
+
+help:
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z0-9_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+build_model: ## build Docker image with compiled model code for serialization
+	cd $(FORTRAN_DIR) && \
+		BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" \
+		COMPILED_IMAGE=$(COMPILED_IMAGE) \
+		COMPILE_OPTION="GT4PY_DEV=Y" \
+		$(MAKE) build_compiled
+
```

Comment:
One thing I'm not positive on -- is this going to trigger a rebuild correctly when files change at ../../FV3?

Reply:
Yes, Docker should pick up the fact that the build context (including the F90 files) have changed. This worked when I introduced the GT4PY_DEV print statement in the code.